test/source_of_truth_test.dart

```diff
@@ -24,5 +24,47 @@ void main() {
       expect(oneResult, equals([1, 2]));
       expect(twoResult, equals([null, 3, 4]));
     });
+
```

Comment:
[General test comments] It would be nice if in the Stock test file you can add a couple of tests to test that if you call clear, the delete method is invoked. Same for the clear all.

Here you have an example of that:
```dart
    test('Stock Clear invokes SOT clear', () async {
      final fetcher = MockFutureFetcher<int, int>();
      var timesCalled = 0;
      when(fetcher.factory).thenReturn((key) => Stream.value(++timesCalled));
      final sourceOfTruth = createMockedSourceOfTruth<int, int>();
      final stock = Stock<int, int>(
        fetcher: fetcher,
        sourceOfTruth: sourceOfTruth,
      );

      await stock.clear(1);
      verify(sourceOfTruth.delete(1)).called(1);
    });
```